Rename webjars-locator to web-dependency-locator

phillip-kruger opened the pull request:
As discussed, we rename (and relocate) the webjars-locator to something more generic, like web-dependency-locator, as now we support mvnpm and importmaps

gsmet reviewed (commented):
Thanks! I added a few comments.

ia3andy commented:
This needs to wait, we are discussing using this opportunity to move it to the quarkiverse. https://quarkusio.zulipchat.com/#narrow/stream/191168-core-team/topic/Quarkus.20Web/near/433256889

We are also discussing making it the way (opt-in via extension) to enable serving of mvnpm and webjars which is currently part of vert.x http.

phillip-kruger commented on extensions/web-dependency-locator/deployment/src/main/java/io/quarkus/webdependency/locator/deployment/ImportMapBuildItem.java:
```diff
@@ -1,4 +1,4 @@
-package io.quarkus.webjar.locator.deployment;
+package io.quarkus.webdependency.locator.deployment;
```
This is a brand new Build Item. Not use anywhere but in the extension

phillip-kruger commented:
@gsmet - please check again, I addressed your comments.
@ia3andy - It's not this extension that allows loading from META-INF. This extension only maps the versions path to the non-versioned path, and now also produce a importmap. 

We can probably move that logic to this extension. I would assume it's in vertx-http at the moment. That would not change if this extension is in Quarkiverse or here.

My suggestion is we merge here and then see if it will be possible to move that logic out of vertx-http and what the effort and impact would be.

phillip-kruger commented:
Things that will break is swagger-ui and graphql-ui.

phillip-kruger committed:
Rename webjars-locator to web-dependency-locator

Signed-off-by: Phillip Kruger <phillip.kruger@gmail.com>

ia3andy commented:
> Things that will break is swagger-ui and graphql-ui.

We might be able to keep the compat through builditems if we keep the logic in vertx http, but add the enablement through builditems.

This way the web-dependency extension can use this SPI for all the detected web dependencies and swagger-ui and graphql-ui can only add the required ones.

gsmet commented:
Is there a reason why we would want to move it to the Quarkiverse?

It's useful and small and is already there so, except if we have a good reason to move it, I would keep it around.

phillip-kruger commented:
I agree @gsmet

ia3andy commented:
it was suggested by @cescoffier. I think it belongs more in quarkiverse and while renaming that might be a good occasion. I have no strong opinion on it.

cescoffier commented:
The reason is just to reduce the size of the core and provide independent release cycles.

phillip-kruger commented:
@gsmet you make a call. It would be good to get this into 3.10

phillip-kruger commented:
Can this go into a patch release ? 3.10.1 ?

gsmet commented:
> Can this go into a patch release ? 3.10.1 ?

No we can't do that. It will have to wait for the next one.
